File: templates/python/v1/designtemplates/factory/shape_factory.py
from abc import ABC, abstractmethod

UNKNOWN_SHAPE_PREFIX = "Unknown shape: "

class Shape(ABC):
    @abstractmethod
    def draw(self) -> str:
        raise NotImplementedError


class Circle(Shape):
    def draw(self) -> str:
        return "Drawing a Circle"


class Square(Shape):
    def draw(self) -> str:
        return "Drawing a Square"


class ShapeFactory:
    @staticmethod
    def create(kind: str) -> Shape:
        normalized_kind = kind.lower()
        match normalized_kind:
            case "circle":
                return Circle()
            case "square":
                return Square()
            case _:
                raise ValueError(f"{UNKNOWN_SHAPE_PREFIX}{kind}")

File: templates/java/v1/designtemplates/factory/ShapeFactory.java
package designtemplates.factory;

public class ShapeFactory {
    private static final String SHAPE_CIRCLE = "circle";
    private static final String SHAPE_SQUARE = "square";
    private static final String UNKNOWN_SHAPE_PREFIX = "Unknown shape: ";

    public static Shape create(String kind) {
        String normalizedKind = kind.toLowerCase();
        return switch (normalizedKind) {
            case SHAPE_CIRCLE -> new Circle();
            case SHAPE_SQUARE -> new Square();
            default -> throw new IllegalArgumentException(UNKNOWN_SHAPE_PREFIX + kind);
        };
    }
}